backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/work/impl/ExecuteQueryLuceneWork.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.lucene.work.impl;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+
+import org.hibernate.search.v6poc.backend.lucene.logging.impl.Log;
+import org.hibernate.search.v6poc.backend.lucene.search.query.impl.LuceneSearcher;
+import org.hibernate.search.v6poc.search.SearchResult;
+import org.hibernate.search.v6poc.util.spi.Futures;
+import org.hibernate.search.v6poc.util.spi.LoggerFactory;
+
+/**
+ * @author Guillaume Smet
+ */
+public class ExecuteQueryLuceneWork<T> implements LuceneQueryWork<SearchResult<T>> {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final LuceneSearcher<T> searcher;
+
+	public ExecuteQueryLuceneWork(LuceneSearcher<T> searcher) {
+		this.searcher = searcher;
+	}
+
+	@Override
+	public CompletableFuture<SearchResult<T>> execute(LuceneQueryWorkExecutionContext context) {
+		return Futures.create( () -> CompletableFuture.completedFuture( executeQuery( searcher ) ) );
```

Comment:
Same as Index works, the fact that this method is blocking should really be mentioned in a "FIXME" comment.